recipes/cpython/all/test_package/conanfile.py

```diff
@@ -0,0 +1,181 @@
+            else:
+                self.output.warn("Module '{}' does not work, but should have worked".format(module))
+                raise exception
+
+    def _cpython_option(self, name):
+        try:
+            return getattr(self.options["cpython"], name, False)
+        except ConanException:
+            return False
+
+    def test(self):
+        if not tools.cross_building(self.settings, skip_x64_x86=True):
+            self.run("{} -c \"print('hello world')\"".format(self.deps_user_info["cpython"].python), run_environment=True)
+
+            buffer = StringIO()
+            self.run("{} -c \"import sys; print('.'.join(str(s) for s in sys.version_info[:3]))\"".format(self.deps_user_info["cpython"].python), run_environment=True, output=buffer)
+            self.output.info(buffer.getvalue())
+            version_detected = buffer.getvalue().splitlines()[-1].strip()
+            if self._py_version != version_detected:
+                raise ConanException("python reported wrong version. Expected {exp}. Got {res}.".format(exp=self._py_version, res=version_detected))
+
+            if self._supports_modules:
+                self._test_module("gdbm", self._cpython_option("with_gdbm"))
+                self._test_module("bz2", self._cpython_option("with_bz2"))
+                self._test_module("bsddb", self._cpython_option("with_bsddb"))
+                self._test_module("lzma", self._cpython_option("with_lzma"))
+                self._test_module("tkinter", self._cpython_option("with_tkinter"))
+                with tools.environment_append({"TERM": "ansi"}):
+                    self._test_module("curses", self._cpython_option("with_curses"))
+
+                self._test_module("expat", True)
+                self._test_module("sqlite3", True)
+                self._test_module("decimal", True)
+                self._test_module("ctypes", True)
+
+            if tools.is_apple_os(self.settings.os) and not self.options["cpython"].shared:
+                self.output.info("Not testing the module, because these seem not to work on apple when cpython is built as a static library")
+                # FIXME: find out why cpython on apple does not allow to use modules linked against a static python
+            else:
+                if self._supports_modules:
+                    with tools.environment_append({"PYTHONPATH": [os.path.join(self.build_folder, "lib")]}):
+                        self.output.info("Testing module (spam) using cmake built module")
+                        self._test_module("spam", True)
+
+                    with tools.environment_append({"PYTHONPATH": [os.path.join(self.build_folder, "lib_setuptools")]}):
+                        self.output.info("Testing module (spam) using setup.py built module")
+                        self._test_module("spam", True)
+
+        if self.options["cpython"].env_vars:
+            self.run(os.path.join("bin", "test_package"), run_environment=True)
```

Comment:
On my local win10 setting of `PYTHONHOME` fixes test failures.

```suggestion
            with tools.environment_append({"PYTHONHOME": os.path.join(self.deps_cpp_info["cpython"].rootpath, "bin")}):
                self.run(os.path.join("bin", "test_package"), run_environment=True)
```